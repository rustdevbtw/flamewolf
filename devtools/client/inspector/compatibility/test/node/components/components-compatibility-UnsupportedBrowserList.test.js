/* Any copyright is dedicated to the Public Domain.
   http://creativecommons.org/publicdomain/zero/1.0/ */

"use strict";

/**
 * Unit tests for the BrowserList component.
 */

const { shallow } = require("enzyme");
const React = require("react");

const UnsupportedBrowserList = React.createFactory(
  require("resource://devtools/client/inspector/compatibility/components/UnsupportedBrowserList.js")
);

describe("UnsupportedBrowserList component", () => {
  it("renders the browsers", () => {
    const list = shallow(
      UnsupportedBrowserList({
        browsers: [
          { id: "firefox", name: "Firefox", version: "69", status: "beta" },
          { id: "firefox", name: "Firefox", version: "70", status: "nightly" },
          { id: "test-browser", name: "Test Browser", version: "1" },
          { id: "test-browser", name: "Test Browser", version: "2" },
          { id: "sample-browser", name: "Sample Browser", version: "100" },
        ],
      })
    );
    expect(list).toMatchSnapshot();
  });

  it("does not show ESR version if newer version is not supported", () => {
    const list = shallow(
      UnsupportedBrowserList({
        browsers: [
          { id: "firefox", name: "Firefox", version: "102", status: "esr" },
          { id: "firefox", name: "Firefox", version: "112", status: "current" },
        ],
      })
    );
    expect(list).toMatchSnapshot();
  });

  it("shows ESR version if newer version is supported", () => {
    const list = shallow(
      UnsupportedBrowserList({
        browsers: [
          { id: "firefox", name: "Firefox", version: "102", status: "esr" },
          { id: "test-browser", name: "Test Browser", version: "1" },
        ],
      })
    );
    expect(list).toMatchSnapshot();
  });
});
