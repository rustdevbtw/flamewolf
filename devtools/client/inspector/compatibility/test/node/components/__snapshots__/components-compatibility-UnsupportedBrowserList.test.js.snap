// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnsupportedBrowserList component does not show ESR version if newer version is not supported 1`] = `
<ul
  className="compatibility-unsupported-browser-list"
>
  <UnsupportedBrowserItem
    id="firefox"
    key="firefox"
    name="Firefox"
    unsupportedVersions={
      Array [
        Object {
          "status": "esr",
          "version": "102",
        },
        Object {
          "status": "current",
          "version": "112",
        },
      ]
    }
    version="112"
  />
</ul>
`;

exports[`UnsupportedBrowserList component renders the browsers 1`] = `
<ul
  className="compatibility-unsupported-browser-list"
>
  <UnsupportedBrowserItem
    id="firefox"
    key="firefox"
    name="Firefox"
    unsupportedVersions={
      Array [
        Object {
          "status": "beta",
          "version": "69",
        },
        Object {
          "status": "nightly",
          "version": "70",
        },
      ]
    }
    version="69"
  />
  <UnsupportedBrowserItem
    id="test-browser"
    key="test-browser"
    name="Test Browser"
    unsupportedVersions={
      Array [
        Object {
          "status": undefined,
          "version": "1",
        },
        Object {
          "status": undefined,
          "version": "2",
        },
      ]
    }
    version="1"
  />
  <UnsupportedBrowserItem
    id="sample-browser"
    key="sample-browser"
    name="Sample Browser"
    unsupportedVersions={
      Array [
        Object {
          "status": undefined,
          "version": "100",
        },
      ]
    }
    version="100"
  />
</ul>
`;

exports[`UnsupportedBrowserList component shows ESR version if newer version is supported 1`] = `
<ul
  className="compatibility-unsupported-browser-list"
>
  <UnsupportedBrowserItem
    id="firefox"
    key="firefox"
    name="Firefox"
    unsupportedVersions={
      Array [
        Object {
          "status": "esr",
          "version": "102",
        },
      ]
    }
    version="102"
  />
  <UnsupportedBrowserItem
    id="test-browser"
    key="test-browser"
    name="Test Browser"
    unsupportedVersions={
      Array [
        Object {
          "status": undefined,
          "version": "1",
        },
      ]
    }
    version="1"
  />
</ul>
`;
